croptalk/evaluate_overall_performance.py

```diff
@@ -0,0 +1,36 @@
+import os
+
+os.chdir('/app')
+
+from langchain.smith import RunEvalConfig
+from langsmith import Client
+
+from croptalk.model_openai_functions import model as MODEL
+
+from dotenv import load_dotenv
+
+load_dotenv('secrets/.env.secret')
+load_dotenv('secrets/.env.shared')
+
+DATASET_NAME = "sp-cp-bp-cih-70"
+
+# DOCS: 
+# https://api.python.langchain.com/en/latest/smith/langchain.smith.evaluation.config.RunEvalConfig.html
+
+eval_config = RunEvalConfig(
+    evaluators=["qa"],
+    reference_key="Answer",  # key in the dataset to compare against
+    input_key="question",  # traced run’s inputs dictionary
+)
+
+client = Client()
+results = client.run_on_dataset(
+    dataset_name=DATASET_NAME,
+    llm_or_chain_factory=MODEL,
+    evaluation=eval_config,
+    input_mapper=lambda x: {"question": x["Question"], 
+                            "chat_history": []},
+    # project_name="CropTalk AprDemo", # has to be unique
+    # tags=["simple_chain"],           # list of tags 
```

Comment:
Out of curiosity, what are these commented lines? When should we uncomment them & what should we set them to?

Reply:
By default, it's better to keep them commented. But if we want to create a unique interpretable experiment name, we can set those. It's just that the run will break if they are uncommented, but not attended to (at least the project name).